packages/shared/src/components/ShareOptionsMenu.tsx

```diff
@@ -41,18 +41,20 @@ export default function ShareOptionsMenu({
   onHidden,
   contextId = 'share-context',
 }: ShareOptionsMenuProps): ReactElement {
-  const link = post && post?.commentsPermalink;
+  const link = post?.commentsPermalink;
   const [, copyLink] = useCopyPostLink(link);
   const { trackEvent } = useContext(AnalyticsContext);
+  const { getShortUrl } = useGetShortUrl();
   const onClick = (provider: ShareProvider) =>
     trackEvent(
       postAnalyticsEvent('share post', post, {
         extra: { provider, origin: Origin.ShareBar },
       }),
     );
 
-  const trackAndCopyLink = () => {
-    copyLink();
+  const trackAndCopyLink = async () => {
+    const shortLink = await getShortUrl(link, 'share_post');
```

Comment:
This is the CID right `share_post` can we maybe store is in some const?
As i've seen it used twice already could use with abstraction layer so it's easier to maintain.

Reply:
Yup, good idea. We will still send it as a string over to the API though, so that we can easily extend in the future in one place only.

Reply:
Yeah 100% 🙏